routes/index.js

```diff
@@ -7,6 +7,8 @@ router.get("/", (req, res) => {
 
 router.get("/problems/all", (req, res) => {
     res.render("problem_set");
+    const ProblemsModel = require("../models/problems");
+    ProblemsModel.find({isVisible: 1},{id: -1,name: 1,tags: 1,difficulty: 1}).sort({id: -1}).then(doc => console.log(doc));
```

Comment:
Now, the data is being retrieved correctly, although it should be `qID` instead of `id`. `id` is the automatically generated for each entry. Also, add `.catch()` after `.then()` to handle the errors.
Now to pass `doc` to the page. You bring the `res.render("problem_set", {data: doc})` inside the `.then()`, after then you can access the data as a variable `data` in the `problem_set.js`

Reply:
@dipanshu231099 make the above changes also.